common/perf-test-core/src/main/java/com/azure/perf/test/core/PerfStressProgram.java

```diff
@@ -249,7 +258,19 @@ public static void runTests(PerfTestBase<?>[] tests, boolean sync, int parallel,
                 forkJoinPool.invokeAll(operations);
 
                 forkJoinPool.awaitQuiescence(durationSeconds + 1, TimeUnit.SECONDS);
-            } else {
+            } else if (completableFuture) {
+                for (PerfTestBase<?> test : tests) {
+                    test.runAllAsyncWithCompletableFuture(endNanoTime).get();
```

Comment:
Isn't this running the `tests` sequentially as calling `get` blocks until the first test running completes? After that aren't all tests effectively no-ops as the `CompleteableFuture` only completes once the test duration completes.

Reply:
yeah, this would have resulted in No-ops after the first execution. 
Updated for each test to be initiated as an independent CompletableFuture, and added to a list to complete parallely but still wait for all to be completed.